onnxruntime/core/optimizer/propagate_cast_ops.cc

```diff
@@ -64,76 +64,73 @@ static std::string GetName(const std::pair<const NodeArg*, std::vector<Node*>>&
+ *  The following two maps specify the opcode to input and opcode to output mappings to list the inputs/outputs to consider while propagating
+ *  cast operations. All other inputs/outputs not listed in this table are not relevant for deciding whether an operation
+ *  performed in float or float16. If an opcode is not listed in these tables, the code will look at all the inputs and outputs to validate
+ *  transformation.
+ */
+static const std::unordered_map<std::string, std::vector<int>> opcode_to_input_map = {
     {"Gather", {0}},
     {"Reshape", {0}},
     {"Dropout", {0}},
     {"Expand", {0}},
     {"LayerNormalization", {0, 1, 2}},
     {"Squeeze", {0}},
-    {"Unsqueeze", {0}}
-};
+    {"Unsqueeze", {0}}};
 
-static std::unordered_map<std::string, std::vector<int>> opcode_to_output_map = {
+static const std::unordered_map<std::string, std::vector<int>> opcode_to_output_map = {
     {"Gather", {0}},
     {"Reshape", {0}},
     {"Dropout", {0}},
     {"Expand", {0}},
     {"LayerNormalization", {0}},
     {"Squeeze", {0}},
-    {"Unsqueeze", {0}}
-};
-
-static std::unordered_set<std::string> inserted_node_names;   // Names of the nodes inserted
-static std::unordered_set<std::string> converted_node_names;  // Names of the nodes converted to FP16
+    {"Unsqueeze", {0}}};
 
 /*
-*  Check if the input is relevant to consider for cast propagation for the given node.
-*  Return true if the opcode is not found in the opcode_to_input map.
-*/
+ *  Check if the input is relevant to consider for cast propagation for the given node.
+ *  Return true if the opcode is not found in the opcode_to_input map.
+ */
 static bool IsRelevantInput(const Node* node, const NodeArg* input) {
-  if (opcode_to_input_map.find(node->OpType()) != opcode_to_input_map.end()) {
-    const std::vector<int>& selected_inputs = opcode_to_input_map[node->OpType()];
+  if (opcode_to_input_map.find(node->OpType()) != opcode_to_input_map.cend()) {
+    const std::vector<int>& selected_inputs = opcode_to_input_map.at(node->OpType());
     int input_index = optimizer_utils::IndexOfNodeInput(*node, *input);
     return std::find(selected_inputs.begin(), selected_inputs.end(), input_index) != selected_inputs.end();
   }
   return true;
 }
 
 /*
-*  Check if the output is relevant to consider for cast propagation for the given node.
-*  Return true if the opcode is not found in the opcode_to_output map.
-*/
+ *  Check if the output is relevant to consider for cast propagation for the given node.
+ *  Return true if the opcode is not found in the opcode_to_output map.
+ */
 static bool IsRelevantOutput(const Node* node, const NodeArg* output) {
-  if (opcode_to_output_map.find(node->OpType()) != opcode_to_output_map.end()) {
-    const std::vector<int>& selected_outputs = opcode_to_output_map[node->OpType()];
+  if (opcode_to_output_map.find(node->OpType()) != opcode_to_output_map.cend()) {
+    const std::vector<int>& selected_outputs = opcode_to_output_map.at(node->OpType());
```

Comment:
double map lookup of `find` and `at` can be avoided by using the result of `find`.